### PR TITLE
Feat/dump filter spaces

### LINKED ISSUE #867: dump filter needs spaces parameter for better readability
- add spaces parameter to the dump template filter
- add tests to dump template filter
- update documentation dump filter

I assign myself to this task

### PR DESCRIPTION
**Pull request by JesseE**
## Summary

Proposed change:

Nunjucks has a convenient `dump` filter which stringifies JSON. But the output is a hard to read string.

This PR proposes a change to the `dump` filter to accept an optional `space` argument and pass that through to `JSON.stringify(input, null, space)`. You can use `|dump(2)` or `|dump('\t')` which gives much more meaningful output.

closes #867

## Checklist

I've completed the checklist below to ensure I didn't forget anything. This makes reviewing this PR as easy as possible for the maintainers. And it gets this change released as soon as possible.

* [x] Proposed change helps towards [*purpose of this project*](https://github.com/mozilla/nunjucks/blob/master/CONTRIBUTING.md#purpose).
* [x] [*Documentation*](https://github.com/mozilla/nunjucks/tree/master/docs/) is added / updated to describe proposed change.
* [x] [*Tests*](https://github.com/mozilla/nunjucks/tree/master/tests) are added / updated to cover proposed change.
* [x] [*Changelog*](https://github.com/mozilla/nunjucks/blob/master/CHANGELOG.md) has an entry for proposed change (if user-facing fix or feature).

### CODE REVIEW

File: docs/templating.md [hunk 1]
```diff
@@ -972,7 +973,69 @@ a b c d e f
 ### dump (object)
```
Comment (jbmoelker) [inline docs/templating.md]: signature changed from `dump (object)` to `dump (object, [space])`, but I think in any case simply `dump` would be better as heading.

File: docs/templating.md [hunk 2]
```diff
@@ -972,7 +973,69 @@ a b c d e f
...
 
 Call `JSON.stringify` on an object and dump the result into the
-template. Useful for debugging: `{{ foo | dump }}`.
+template. Useful for debugging: `{{ items | dump }}`. 
+
+**Input**
+
+```jinja
+{% set items = [{ a : 1}, { b : false}] %}
```
Comment (jbmoelker) [inline docs/templating.md]: I think you want to illustrate `dump` works on different primitives, so I'd use `['a', 1, { b : false}]` in all examples in the docs. The only thing you'd change between examples is the `space` parameter (none, 2 spaces, 1 tab).

File: tests/filters.js [hunk 3]
```diff
@@ -86,6 +86,18 @@
             finish(done);
         });
 
+        it('dump', function(done) {
+            equal('{{ {"container":[{\'a\': true}, {\'b\': 2 }]} | dump }}',
```
Comment (jbmoelker) [inline tests/filters.js]: Again I think you want to illustrate dump works on different primitives, so I'd use `['a', 1, { b : false}]` (same as in doc examples) in all tests. The only thing you'd change between test cases is the space parameter (none, 2 spaces, 1 tab).

**Comment by jbmoelker** [comment]
@JesseE looks good to me! I'd like to get this change in. The added filesize is neglectable (so doesn't hurt browser use). @vecmezoni do you agree?

**Comment by vecmezoni** [comment]
Thank you!

**Pull request merged**